### ISSUE TITLE
OWS fails load JNI library from jar file on macos

### ISSUE THREAD

**Issue by berniegsfc** [karakun/OpenWebStart#323]
The application at https://spdf.gsfc.nasa.gov/skteditor/skteditor.jnlp depends upon a JNI library that is contained in a jar specified in https://spdf.gsfc.nasa.gov/skteditor/cdf/cdf-latest.jnlp.  While the application runs on windows (OWS) and linux, it fails with OWS (bundled azul 11.0.7 jvm) on macos 10.15.6.  The OWS console shows the following message "no cdfNativeLibrary in java.library.path: [/Users/btharris/Library/Java/Extensions, /Library/Java/Extensions, /Network/Library/Java/Extensions, /System/Library/Java/Extensions, /usr/lib/java, .]".  The jar file containing the jni file was downloaded at
> $ jar -tvf .cache/icedtea-web/cache/0/3/nativeCdfLibs.jar      
>   ...
> 907512 Wed Jul 15 10:45:26 EDT 2020 libcdfNativeLibrary.jnilib
OWS on macos should load jni libraries from jar files like it does on windows (and Oracle's WebStart did in Java 8).

**Comment by AndreasEhret** [comment]
I could not access the JNLP file by the links you provided to test with my Mac. Can you check that the URL provided is accessible and that there is a possibility to have a look into the JNLP?

**Comment by berniegsfc** [comment]
I don't understand why you cannot access those urls.  That is a public site.  Can you access https://spdf.gsfc.nasa.gov/?  What error message do you get.

**Comment by AndreasEhret** [comment]
Ok, sorry found it, was related to my VPN. Now it works and I can have a look at it.

**Comment by berniegsfc** [comment]
Thanks for looking into this and let me know if you have any further problems/questions.

**Comment by AndreasEhret** [comment]
I can reproduce and finally found the cause. It is related to https://bugs.openjdk.java.net/browse/JDK-8127215 and https://bugs.java.com/bugdatabase/view_bug.do?bug_id=7158157. Your application uses still the older naming with "lib" prefix and .jnilib extension for the native lib file. 

While that was okay in Java 6, for compatibility with the BSD port this was changed to .dylib extension and this is what OWS/IcedTea-Web was looking for (and couldn't find in your example).

I've created a pull request with a fix for this in IcedTea-Web https://github.com/AdoptOpenJDK/IcedTea-Web/pull/695

This fix will be available in the next OWS release in November.
